lib/msal-angular/src/msal-guard.service.ts

```diff
@@ -50,21 +49,22 @@ export class MsalGuard implements CanActivate {
         }
         //token is expired/deleted but userdata still exists in _oauthData object
         else if (!this.authService._oauthData.isAuthenticated && this.authService._oauthData.userName) {
-            return new Promise((resolve, reject) => {
-                this.authService.acquireTokenSilent([this.config.clientID]).then((token: any) => {
+            return this.authService.acquireTokenSilent([this.config.clientID])
+                .then((token: any) => {
                     if (token) {
                         this.authService._oauthData.isAuthenticated = true;
                         var authenticationResult = new AuthenticationResult(token );
                         this.broadcastService.broadcast("msal:loginSuccess",  authenticationResult);
-                        resolve (true);
+                        return true
```

Comment:
semicolon missing. I know JS doesn't care but good for syntactic uniformity across the code